File: SDKs/amazonaws.com/glue/2017-03-31/python/src/sdk/models/shared/createregistryinput.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class CreateRegistryInput:
    registry_name: str = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('RegistryName') }})
    description: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Description') }})
    tags: Optional[dict[str, str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    

File: SDKs/amazonaws.com/glue/2017-03-31/typescript/src/sdk/models/shared/createregistryinput.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class CreateRegistryInput extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=RegistryName" })
  registryName: string;

  @SpeakeasyMetadata({ data: "json, name=Tags" })
  tags?: Map<string, string>;
}
